src/checks.py

```diff
@@ -312,12 +313,36 @@ def check_git_revision(state: State) -> Optional[str]:
 def check_blacklisted_files(state: State) -> Optional[str]:
     blacklist = [".git", ".gitignore", ".mvn", "mvnw", "mvnw.cmd", "Jenkinsfile"]
     commands = [
-        f"test $(find {state.source_dir} -name {item} | wc -l) -eq 0"
+        f"find {state.source_dir} -name {item} | ifne bash -c 'cat && false'"
         for item in blacklist
     ]
     return _check_sh(commands)
 
 
+@check("No .gitignore-d files in git checkout", hide_if_passing=True)
+def check_gitignore_in_repo(state: State) -> Optional[str]:
+    return _check_sh("shopt -s globstar; ! git check-ignore **", workdir=state.git_dir)
+
+
+@check("No .gitignore-d files in source archive", hide_if_passing=False)
+def check_gitignore_in_release(state: State) -> Optional[str]:
+    error = _check_sh(
+        f"find . -name .gitignore | xargs cp --parents -t {state.source_dir}",
+        workdir=state.git_dir,
+    )
+    if not error:
```

Comment:
Notice how I resisted the temptation to build a monad abstraction around `Optional[str]` and `_check_sh`.